src/lib/mcp.ts

```
@@ -0,0 +1,316 @@
+        mcpObservability.emit('mcp:tools:fetch:start');
+
+        try {
+            const result = await client.listTools();
+            let toolsMap: Record<string, Tool> = {};
+
+            for (const tool of result.tools) {
+                // Apply filters
+                if (options?.include && !options.include.some(pattern => tool.name.includes(pattern))) {
+                    continue;
+                }
+                if (options?.exclude && options.exclude.some(pattern => tool.name.includes(pattern))) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)